### ISSUE TITLE
(3.0.0-3.6.1) ParallelCluster CLI installation could fail on Python 3.10

### ISSUE THREAD

**Issue by jdeamicis** [aws/aws-parallelcluster#5517]
## Bug description

PyYAML is a Python dependency of ParallelCluster. [Cython](https://pypi.org/project/Cython/#history) is a Python dependency of PyYAML. Due to a recent release of Cython version 3, the pip install aws-parallelcluster command fails on Python>=3.10 if PyYAML 5.4.1 is drawn in as a dependency.

```
$ pip install aws-parallelcluster
...
Collecting PyYAML~=5.3
  Using cached PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
...        
        File "/private/var/folders/rc/026kj4vj1nnbzzkm7z193tfc0000gr/T/pip-build-env-w9dt3icw/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
```

For more information, see https://github.com/yaml/pyyaml/issues/724

## Mitigation

Install PyYAML pinned to version 5.3.1 when installing ParallelCluster:

```
pip install PyYAML==5.3.1 aws-parallelcluster
```

**Comment by ccassidy-vaisala** [comment]
FYI later in that PyYAML issue thread there is mention of a vulnerability in versions prior to 5.4.  I see PyYAML 6.0 was released, is it possible to bump the dependency version up to 6?

**Comment by chenwany** [comment]
> FYI later in that PyYAML issue thread there is mention of a vulnerability in versions prior to 5.4. I see PyYAML 6.0 was released, is it possible to bump the dependency version up to 6?

Thank you for the information!
Yes, that should work too, in the develop branch we already bump PyYAML to 6.0.1 See https://github.com/aws/aws-parallelcluster/pull/5509

------
Update: pin to 6.0.1 doesn't work for old versions of Parallelcluster since it requires depends on PyYAML~=5.3

**Comment by enrico-usai** [comment]
Closed, since we released ParallelCluster 3.7.0 where we bumped PyYAML version to 6.x.
